qualtran/bloqs/arithmetic/addition.py

```diff
@@ -359,6 +361,63 @@ def _cvs_converter(vv):
     return tuple(int(v) for v in vv)
 
 
+@frozen
+class XorK(Bloq):
+    r"""Maps |x> to |x \oplus k> for a constant k.
+
+    Args:
+        dtype: Data type of the input register `x`.
+        k: The classical integer value to be XOR-ed to x.
+        cvs: A tuple of control values. Each entry specifies whether that control line is a
+            "positive" control (`cv[i]=1`) or a "negative" control (`cv[i]=0`).
+
+    Registers:
+        x: A quantum register of type `self.dtype` (see above).
+        ctrls: A sequence of control qubits (only when `cvs` is non-empty).
+    """
+    dtype: QDType
+    k: SymbolicInt
```

Comment:
If `k` is a `SymbolicInt` should we restrict `dtype` to be `QUInt` / `QInt` ? Looking at the implementation below, it seems like this would also work for other types when `k` is not necessarily an integer.